pkgs/development/tools/godot/3/default.nix

```diff
@@ -1,108 +1,135 @@
-{ stdenv
-, lib
+{ stdenv, lib
+, alsa-lib
+, alsa-plugins
 , fetchFromGitHub
-, scons
-, pkg-config
-, udev
+, freetype
+, installShellFiles
+, libGLU
+, libpulseaudio
 , libX11
 , libXcursor
+, libXext
+, libXfixes
+, libXi
 , libXinerama
 , libXrandr
 , libXrender
-, libpulseaudio
-, libXi
-, libXext
-, libXfixes
-, freetype
-, openssl
-, alsa-lib
-, alsa-plugins
 , makeWrapper
-, libGLU
-, zlib
+, openssl
+, pkg-config
+, scons
+, udev
 , yasm
-, withUdev ? true
+, zlib
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self: {
   pname = "godot";
   version = "3.5.1";
 
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
+    rev = "${self.version}-stable";
     sha256 = "sha256-uHwTthyhfeQN0R1XjqZ+kGRa5WcpeQzA/DO9hZk4lvU=";
   };
 
-  nativeBuildInputs = [ pkg-config makeWrapper ];
-  buildInputs = [
+  nativeBuildInputs = [
+    installShellFiles
+    pkg-config
+    makeWrapper
     scons
-    udev
+  ];
+
+  buildInputs = [
+    alsa-lib
+    freetype
+    libGLU
+    libpulseaudio
     libX11
     libXcursor
+    libXext
+    libXfixes
+    libXi
     libXinerama
     libXrandr
     libXrender
-    libXi
-    libXext
-    libXfixes
-    freetype
     openssl
-    alsa-lib
-    libpulseaudio
-    libGLU
-    zlib
+    udev
     yasm
+    zlib
   ];
 
-  patches = [ ./pkg_config_additions.patch ./dont_clobber_environment.patch ];
```

Comment:
I can't test building on MacOS, and only have tested on NixOS.  It builds within a nix develop environment as well as with nix build.  It's possible there's some context in which it fails without that patch but I don't know what that would be.

I simply eliminated the patch because it built fine without it, and thus has no clear benefit.  Maybe there is a more subtle benefit to it, but i don't like to keep code around unless its purpose is actually known.  Otherwise it just lingers indefinitely like an appendix...

Do either of you feel strongly that we should keep the patch?  And if so, what should I say it fixes, when i write the documentation for it?

Reply:
I'd say that if you strongly tested and believe that it does not change anything, to remove it because it increases the maintenance burden. I was mostly offhand commenting about winter's take on that issue!

Reply:
Per [comment](https://github.com/NixOS/nixpkgs/pull/177578#discussion_r1110392620) — I would expect removing this patch to break, or cause non-isolated behaviour/non-deterministic builds, when a binary or pkgconfig configuration used by Godot or the Godot build is present in the host system's magic paths (e.g. `/usr/bin`) — and furthermore one that isn't a library, as these will be stripped out by `NIX_ENFORCE_PURITY` in the majority of cases where that's enabled.  Except in this rather specific edge case, you probably won't see breakage.